Clarify docs for rendering nothing in child and attribute expressions (#908)
lit/lit.dev: packages/lit-dev-content/site/docs/components/shadow-dom.md

<div class="alert alert-info">

> **Rendering fallback content.** If any child nodes are assigned to a slot, its fallback content doesn't render. A default slot with no name accepts any child nodes. It won't render fallback content even if the only assigned nodes are text nodes containing whitespace, for example `<example-element> </example-element>`. When using a Lit expression as a child of an element with Shadow DOM and a slot with fallback content, make sure to use a non-rendering value when appropriate so that fallback content is rendered. See [non-rendering values](/docs/templates/expressions/#non-rendering) for more information.

Reviewer: The sentence is pretty dense, I think you can elide a bit of detail because "so that any slot fallback content" implies a CE with a shadow DOM with a slot with default content...
```suggestion
**Rendering fallback content.** If any child nodes are assigned to a slot, its fallback content doesn't render. A default slot with no name accepts any child nodes. It won't render fallback content even if the only assigned nodes are text nodes containing whitespace, for example `<example-element> </example-element>`. When using a Lit expression as a child of a custom element, make sure to use a non-rendering value when appropriate so that any slot fallback content is rendered. See [non-rendering values](/docs/templates/expressions/#non-rendering) for more information.
```
